pandas/core/arrays/base.py

```diff
@@ -667,7 +681,12 @@ def argmax(self, skipna: bool = True) -> int:
             raise NotImplementedError
         return nargminmax(self, "argmax")
 
-    def fillna(self, value=None, method=None, limit=None):
+    def fillna(
+        self: ExtensionArray,
```

Comment:
FYI: no need to update at the moment. if #36092 goes in first, you'll probably need to update this PR anyway.

`self` does not need a type annotation except when using a typevar.

Reply:
see https://mypy.readthedocs.io/en/stable/generics.html?highlight=self#generic-methods-and-generic-self and mypy 0.740 release notes http://mypy-lang.blogspot.com/search?updated-max=2019-11-29T11:39:00-08:00&max-results=7

> The self argument to methods without annotations is now correctly given the object’s type (instead of Any) when using check_untyped_defs (PR 7530).

Reply:
I just tried removing them all from `pandas/core/arrays/base.py` and that seems to work.